src/lib.rs

```diff
@@ -297,3 +297,17 @@ broker_handler!(CallbackHandler, |app, req| {
         try!(oidc::verify(app, session, code))
     )
 });
+
+
+#[cfg(test)]
+mod tests {
+    // TODO: Refactor so these tests fit somewhere more natural...
```

Comment:
Maybe the resolution here is to leave this open and unmerged until I / we can write some tests covering this sort of thing?